src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const optionsPerPage = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(optionsPerPage[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const firstVisibleItemIndex = (currentPage - 1) * perPage;
+  const lastVisibleItemIndex = Math.min(42, currentPage * perPage);
+
+  const visibleItems = items.slice(firstVisibleItemIndex, lastVisibleItemIndex);
+
+  const handlePageChange = (newPage: number) => {
+    const maxPage = Math.ceil(items.length / perPage);
+
+    if (newPage >= 1 && newPage <= maxPage) {
+      setCurrentPage(newPage);
+    }
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${firstVisibleItemIndex + 1} - ${lastVisibleItemIndex} of ${items.length})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            defaultValue={optionsPerPage[1]}
```

Comment:
Using 'defaultValue' in the select element is not recommended when the select value is controlled by React state. You should use the 'value' prop instead to ensure the selected option is in sync with the component's state.